File: app/src/main/java/vergecurrency/vergewallet/service/model/MnemonicManager.java
package vergecurrency.vergewallet.service.model;

import com.google.gson.Gson;

public class MnemonicManager {


	private String[] mnemonic;

	public static String[] getMnemonicFromJSON(String mnemonic) {
		return new Gson().fromJson(mnemonic, String[].class);
	}

	public String getMnemonicAsJSON() {
		return new Gson().toJson(this.mnemonic);
	}

	public String[] getMnemonic() {
		return mnemonic;
	}

	public void setMnemonic(String[] mnemonic) {
		this.mnemonic = mnemonic;
	}
}

File: app/src/main/kotlin/vergecurrency/vergewallet/service/model/MnemonicManager.kt
package vergecurrency.vergewallet.service.model

import com.google.gson.Gson

class MnemonicManager {

    companion object {

        fun getMnemonicFromJSON(mnemonic: ByteArray): Array<ByteArray>? {
            return Gson().fromJson(String(mnemonic), Array<ByteArray>::class.java)
        }
    }
}
